runtime/interpreter/storage.go

```diff
@@ -64,17 +64,33 @@ func ConvertStoredValue(gauge common.MemoryGauge, value atree.Value) (Value, err
 		if !ok {
 			panic(errors.NewUnreachableError())
 		}
-		return newArrayValueFromConstructor(gauge, staticType, value.Count(), func() *atree.Array { return value }), nil
+		return newArrayValueFromAtreeArray(
+			gauge,
+			staticType,
+			ArrayElementSize(staticType),
+			value,
+		), nil
 
 	case *atree.OrderedMap:
 		typeInfo := value.Type()
-		switch typeInfo := typeInfo.(type) {
+		switch staticType := typeInfo.(type) {
 		case DictionaryStaticType:
-			return newDictionaryValueFromConstructor(gauge, typeInfo, value.Count(), func() *atree.OrderedMap { return value }), nil
+			return newDictionaryValueFromAtreeMap(
+				gauge,
+				staticType,
+				DictionaryElementSize(staticType),
+				value,
+			), nil
+
```

Comment:
we already pass `staticType`, is there a case where count can be different then `DictionaryElementSize(staticType)` ?

Reply:
`newDictionaryValueFromAtreeMap` is the "lowest level" constructor, and requires an element size to be passed. 
It is true that here, it can be derived from the static type.

However, this allows [`newDictionaryValueFromConstructor`](https://github.com/onflow/cadence/pull/2509/files#diff-424575c293dc2bc3a28aab845829d7656fa83a7be4e026de8906935e23bbd52eR16132) to compute the element size once, use it for memory metering, and then pass it to `newDictionaryValueFromAtreeMap` so it can be reused.